code/modules/research/nanites/nanite_programs/protocols.dm

```diff
@@ -98,3 +80,188 @@
+		return
+
+	var/mob/living/carbon/C = host_mob
+	var/obj/item/organ/liver/liver = C.getorganslot(ORGAN_SLOT_LIVER)
+	var/obj/item/organ/stomach/stomach = C.getorganslot(ORGAN_SLOT_STOMACH)
+	var/obj/item/organ/lungs/lungs = C.getorganslot(ORGAN_SLOT_LUNGS)
+	var/obj/item/organ/heart/heart = C.getorganslot(ORGAN_SLOT_HEART)
+	var/obj/item/organ/eyes/eyes = C.getorganslot(ORGAN_SLOT_EYES)
+	var/obj/item/organ/ears/ears = C.getorganslot(ORGAN_SLOT_EARS)
+	var/obj/item/organ/brain/brain = C.getorganslot(ORGAN_SLOT_BRAIN)
+
+	if(nanites.nanite_volume < 500)
+		return
+
+	var/current_stage = 0
+	if(nanites.nanite_volume > 500) //Liver is the main hub of nanite replication and the first to be threatened by excess volume
+		if(prob(10))
+			liver.applyOrganDamage(0.6)
+		current_stage++
+	if(nanites.nanite_volume > 750) //Extra volume spills out in other central organs
+		if(prob(10))
+			stomach.applyOrganDamage(0.75)
+		if(prob(10))
+			lungs.applyOrganDamage(0.75)
+		current_stage++
+	if(nanites.nanite_volume > 1000) //Extra volume spills out in more critical organs
+		if(prob(10))
+			heart.applyOrganDamage(0.75)
+		if(prob(10))
+			brain.applyOrganDamage(0.75)
+		current_stage++
+	if(nanites.nanite_volume > 1250) //Excess nanites start invading smaller organs for more space, including sensory organs
+		if(prob(13))
+			eyes.applyOrganDamage(0.75)
+		if(prob(13))
+			ears.applyOrganDamage(0.75)
+		current_stage++
+	if(nanites.nanite_volume > 1500) //Nanites start spilling into the bloodstream, causing toxicity
+		if(prob(15))
+			C.adjustToxLoss(0.5, TRUE, forced = TRUE) //Not healthy for slimepeople either
+		current_stage++
+	if(nanites.nanite_volume > 1750) //Nanites have almost reached their physical limit, and the pressure itself starts causing tissue damage
+		if(prob(15))
+			C.adjustBruteLoss(0.75, TRUE)
+		current_stage++
+
+	volume_warning(current_stage)
+
+/datum/nanite_program/protocol/unsafe_storage/proc/volume_warning(tier)
+	if(world.time < next_warning)
+		return
+
+	var/list/main_warnings
+	var/list/extra_warnings
+
+	switch(tier)
+		if(1)
+			main_warnings = volume_warnings_stage_1
+			extra_warnings = null
+		if(2)
+			main_warnings = volume_warnings_stage_2
+			extra_warnings = volume_warnings_stage_1
+		if(3)
+			main_warnings = volume_warnings_stage_3
+			extra_warnings = volume_warnings_stage_1 + volume_warnings_stage_2
+		if(4)
+			main_warnings = volume_warnings_stage_4
+			extra_warnings = volume_warnings_stage_1 + volume_warnings_stage_2 + volume_warnings_stage_3
+		if(5)
+			main_warnings = volume_warnings_stage_5
+			extra_warnings = volume_warnings_stage_1 + volume_warnings_stage_2 + volume_warnings_stage_3 + volume_warnings_stage_4
+		if(6)
+			main_warnings = volume_warnings_stage_6
+			extra_warnings = volume_warnings_stage_1 + volume_warnings_stage_2 + volume_warnings_stage_3 + volume_warnings_stage_4 + volume_warnings_stage_5
```

Comment:
Any way to make this pattern cleaner?

Reply:
I'm open to ideas, i've iterated on this block a bit and this is the best i could do so far